### DISCUSSION TITLE
Rustdesk Pro : The connection is not allowed. You have not logged in or your login session has expired.

### DISCUSSION THREAD

**Discussion by rexmercismollis** [rustdesk/rustdesk-server-pro#414] [General]
I#m running self-hosted Rustdesk Pro (1 user, 20 devcies) as docker contianer. Between several devices I can establish connection without any problem. But between sevaral other devices I cannot. I obtained the following message: "VerbindungsfehlerThe connection is not allowed. You have not logged in or your login session has expired." From these devices I can etablish connection to other devices. They must be connected with my self-hosted relay server. What is wrong? I currently use Teamviewer and would like to move to other remote software. This issue possibly prevent me to decide to use rustdesk instead of Teamviewer!
Thank and best regards
-Nghia-

**Comment by rustdesk** [comment]
Log in on controlling side. When your controlled side is assigned to some user (aka permission-enabled devices), the controlling side will be also required to login because of permission control. That's why you saw
> Between several devices I can establish connection without any problem. But between sevaral other devices I cannot.`.

Acutally the message `You have not logged in or your login session has expired` has told you everyting.

https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/permissions/

`login` is also required when you have enabled `concurrent connection limits`.

<img width="1012" alt="Screenshot 2024-10-11 at 1 42 51 PM" src="https://github.com/user-attachments/assets/c17067fb-c52d-4767-a2e0-817f9ba7ccb0">

**Comment by fufesou** [comment]
Please also check this option

![image](https://github.com/user-attachments/assets/fc355b43-3160-46fa-a51d-fc4eaa0ddf98)
